main/acle.md

```diff
@@ -4726,6 +4881,2404 @@ assertions, which may speed up access to aligned data (and will fault
+
+Let VLST be a valid C++ type:
+
+``` c
+  VLAT __attribute__((arm_sve_vector_bits(N)))
+```
+
+for some SVE vector type or SVE predicate type VLAT. VLST is mangled in
+the same way as a template:
+
+``` c
+  template<typename, unsigned> struct __SVE_VLS;
+```
+
+with the arguments:
+
+``` c
+  __SVE_VLS<VLAT, N>
+```
+
+For example:
+
+``` c
+  #if __ARM_FEATURE_SVE_BITS==512
+  // Mangled as 9__SVE_VLSIu11__SVInt32_tLj512EE
+  typedef svint32_t vec __attribute__((arm_sve_vector_bits(512)));
+  // Mangled as 9__SVE_VLSIu10__SVBool_tLj512EE
+  typedef svbool_t pred __attribute__((arm_sve_vector_bits(512)));
+  #endif
+```
+
+## SVE enum declarations
+
+The following enum enumerates all the possible patterns returned by a PTRUE:
+
+``` c
+  enum svpattern
+  {
+    SV_POW2 = 0,
+    SV_VL1 = 1,
+    SV_VL2 = 2,
+    SV_VL3 = 3,
+    SV_VL4 = 4,
+    SV_VL5 = 5,
+    SV_VL6 = 6,
+    SV_VL7 = 7,
+    SV_VL8 = 8,
+    SV_VL16 = 9,
+    SV_VL32 = 10,
+    SV_VL64 = 11,
+    SV_VL128 = 12,
+    SV_VL256 = 13,
+    SV_MUL4 = 29,
+    SV_MUL3 = 30,
+    SV_ALL = 31
+  };
+```
+
+The following enum lists the possible prefetch types:
+
+``` c
+  enum svprfop
+  {
+    SV_PLDL1KEEP = 0,
+    SV_PLDL1STRM = 1,
+    SV_PLDL2KEEP = 2,
+    SV_PLDL2STRM = 3,
+    SV_PLDL3KEEP = 4,
+    SV_PLDL3STRM = 5,
+    SV_PSTL1KEEP = 8,
+    SV_PSTL1STRM = 9,
+    SV_PSTL2KEEP = 10,
+    SV_PSTL2STRM = 11,
+    SV_PSTL3KEEP = 12,
+    SV_PSTL3STRM = 13
+  };
+```
+
+Both enums are defined by [`<arm_sve.h>`](#arm_sve.h).
+
+## SVE intrinsics
+
+### SVE naming convention
+
+The SVE ACLE intrinsics have the form:
+
+``` c
+  svbase[_disambiguator][_type0][_type1]…[_predication]
+```
+
+where the individual parts are as follows:
+
+**base**
+
+> For most intrinsics this is the lower-case name of an SVE
+> instruction, but with some adjustments:
+>
+> *   The most common change is to drop `F`, `S` and `U` if they
+>     stand for “floating-point”, “signed” and “unsigned” respectively,
+>     in cases where this would duplicate information in the type
+>     suffixes below.
+>
+> *   Simple non-extending loads and non-truncating stores drop the
+>     size suffix (`B`, `H`, `W` or `D`), which would always duplicate
+>     information in the suffixes.
+>
+> *   Conversely, extending loads always specify an explicit extension
+>     type, since this information is not available in the suffixes.
+>     A sign-extending load has the same base as the architectural
+>     instruction (e.g. `ld1sb`) while a zero-extending load replaces
+>     the `s` with a `u` (e.g. `ld1ub` for a zero-extending `LD1B`).
+>     Thus [`svld1ub_u32`](https://developer.arm.com/architectures/instruction-sets/intrinsics/#q=svld1ub_u32)
+>     zero-extends 8-bit data to a vector of
+>     `uint32_t`s while [`svld1sb_u32`](https://developer.arm.com/architectures/instruction-sets/intrinsics/#q=svld1sb_u32)
+>     sign-extends 8-bit data to a vector of `uint32_t`s.
```

Comment:
Why are these lines and those following quoted?

Reply:
> Why are these lines and those following quoted?

It's just a way of formatting the text to look like a description list.  There is an extension that supports description lists directly:
```
thing

:   description
```
but although pandoc supports it, github doesn't.

Reply:
Why not make the description a header of level 4? This would look useful in the toc:

## SVE intrinsics
### SVE naming convention
#### base
description of base...
#### disambiguator
...
#### type0 and type1
...
#### predication
...

Reply:
It's actually a nested description list: we have another one inside the “predication” entry, so we'd still need something there.

I don't think “base” on its own would be a good section heading.  It doesn't give much context and it doesn't follow the house style of using sentence case.

Reply:
Oh I see - so it is more complicated than it looks. This seems to be an extra reason for using level sectioning. If `base` and others are not good titles, maybe let's try to use something along the lines of 

```
#### Base convention
#### Type0 and Type1 conventions
#### Convention for predication
##### Zeroing - `z`
....
```

Reply:
But the current “base”, “disambiguator” etc. refer back to:

```
  svbase[_disambiguator][_type0][_type1]…[_predication]
```

Rewording them loses that link.

Also, I think the current version looks better in terms of formatting.  It was a description list in the original doc, and for instance we have:
```
  **predication**
  
  > This suffix describes the inactive elements in the result of a
  > predicated operation. It can be one of the following:
  >
  > **`z`**
  >
  > > Zero predication: set all inactive elements of the result to zero.
  > …
  > Predicated loads, predicated comparisons and predicated string match
  > operations are always zeroing operations, so for brevity, the
  > corresponding intrinsics have no predication suffix.
```
where the final paragraph is part of the general “predication” entry.